datacube/index/memory/_datasets.py

```diff
@@ -597,19 +599,19 @@ def _product_period_count(
             ],
         ]
     ]:
-        YieldType = Tuple[Product, Iterable[Tuple[Range, int]]]
+        yieldtype = Tuple[Product, Iterable[Tuple[Range, int]]]
```

Comment:
I prefer `YieldType` capitalised, as it is standing in for a class/type. 

What flake8 error code was it throwing?